### PR TITLE
NET-5951 - Unique route names for implicit routes

### PR DESCRIPTION
**Pull request by jmurret**
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: internal/mesh/internal/controllers/sidecarproxy/builder/naming.go [hunk 1]
```diff
@@ -28,10 +28,19 @@ func DestinationStatPrefix(serviceRef *pbresource.Reference, portName, datacente
...
 	if port != 0 {
-		return fmt.Sprintf("%s:%s:%s:%d", name, portName, address, port)
+		return fmt.Sprintf("%s:%d", name, port)
 	}
 
-	return fmt.Sprintf("%s:%s:%s", name, portName, address)
+	return name
+}
+
+// XDSResourceID returns a string representation that uniquely identifies the
```
Comment (ndhanushkodi) [inline internal/mesh/internal/controllers/sidecarproxy/builder/naming.go:40]:
```suggestion
// DestinationResourceID returns a string representation that uniquely identifies the
```

File: internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func (b *Builder) buildDestination(
 			}
 		}
 	}
-
+
```
Comment (erichaberkorn) [inline internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go:74]: Nit: whitespace

**Review by erichaberkorn — APPROVED** [review]

**Review by ndhanushkodi — APPROVED** [review]
looks great!!

File: internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go (hunk 2 above)
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go]: thank you. fixed in b9c43cfa5cc6fe3c6b368d8ef94c3ce1a72db341

File: internal/mesh/internal/controllers/sidecarproxy/builder/naming.go (hunk 1 above)
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/naming.go]: ty!  fixed in b9c43cfa5cc6fe3c6b368d8ef94c3ce1a72db341

**Pull request merged**